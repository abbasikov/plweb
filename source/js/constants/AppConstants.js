/*
 * AppConstants
 * These are the variables that determine what our central data store (reducer.js)
 * changes in our state. When you add a new action, you have to add a new constant here
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'YOUR_ACTION_CONSTANT';
 */

// The initial application state
export const INITIAL_STATE = {
    formState: {
        username: '',
        password: ''
    },
    sendingLoginRequest: false,
    loggedIn: false,
    errorMessage: '',
    currentUser:{},
    loginError:'',
    homeScreenLoading:false,
    feeds:[]
};

export const SERVER_URL = 'https://demo5100262.mockable.io';
export const CHANGE_FORM = 'CHANGE_FORM';
export const SET_AUTH = 'SET_AUTH';
export const SET_FEEDS = 'SET_FEEDS';
export const SENDING_REQUEST = 'SENDING_REQUEST';
export const SENDING_LOGIN_REQUEST = 'SENDING_LOGIN_REQUEST';
export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';
export const FETCH_POSTS = 'FETCH_POSTS';
export const CREATE_POST = 'CREATE_POST';
export const FETCH_POST = 'FETCH_POST';
export const DELETE_POST = 'DELETE_POST';
export const SELECT_BOOK = "SELECT_BOOK";
export const LOGIN_USER = "LOGIN_USER";
export const SET_HOME_SCREEN_LOADING = "SET_HOME_SCREEN_LOADING";