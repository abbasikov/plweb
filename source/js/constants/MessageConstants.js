/*
 * MessageConstants
 * These are the variables that contain the text which is displayed on certain errors
 *
 * Follow this format:
 * export const YOUR_CONSTANT = 'Your message text';
 */
export const FIELD_MISSING = 'Please fill out the entire form.';
export const WRONG_PASSWORD = 'Wrong password.';
export const USER_NOT_FOUND = 'This username does not exist.';
export const USERNAME_TAKEN = 'Sorry, but this username is already taken';
export const GENERAL_ERROR = 'Something went wrong, please try again';
export const LOADING_TEXT= "Please wait...";