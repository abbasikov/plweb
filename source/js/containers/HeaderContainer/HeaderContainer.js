import React from 'react';
import $ from 'jquery';
import { connect } from 'react-redux';
import { logoutUser,setHomeScreenLoading } from '../../actions/actions';
import { bindActionCreators } from 'redux';
import { browserHistory } from 'react-router';
import routes from '../../routes';

import ClickOutsideMixin from '../../components/ClickOutsideMixin/ClickOutsideMixin';
import VerticalSlideMenu from '../../components/VerticalSlideMenu/VerticalSlideMenu';

class HeaderContainer extends ClickOutsideMixin {

    static propTypes = {
        currentUser:React.PropTypes.object,
        loggedIn:React.PropTypes.bool
    };


    constructor(props) {
        super(props);
        this.state = {
            courseSwitchOpen: false,
            userSwitchOpen: false,
            mobileSwitchOpen: false,
            messagesSwitchOpen:false
        }
    }

    componentDidMount() {
        super.componentDidMount();
    }

    componentWillUnmount () {
        super.componentWillUnmount();
    }

    componentWillMount() {
    }

    componentWillReceiveProps(nextProps){
    }

    toggleDropdown(name) {
        console.log('toggle called');
        if(name === 'course-switch') {
            this.setState({
                courseSwitchOpen: !this.state.courseSwitchOpen,
                userSwitchOpen:false,
                messagesSwitchOpen:false
            });
        }
        if(name === 'user-switch') {
            this.setState({
                courseSwitchOpen: false,
                userSwitchOpen:!this.state.userSwitchOpen,
                messagesSwitchOpen:false
            });
        }

        if(name === 'mobile-switch') {
            this.setState({
                mobileSwitchOpen:!this.state.mobileSwitchOpen
            });
        }

        if(name === 'messages-switch') {
            this.setState({
                messagesSwitchOpen:!this.state.messagesSwitchOpen,
                userSwitchOpen:false,
                courseSwitchOpen: false
            });
        }
    }

    clickedInside() {
        //console.log('inside');
    }

    clickedOutside() {
        this.setState({
            courseSwitchOpen: false,
            userSwitchOpen:false,
            messagesSwitchOpen:false
        });
    }

    logoutClick(e) {
        e.preventDefault();
        this.closeSlideMenu();
        this.props.logoutUser();
    }

    clickHome(e) {
        e.preventDefault();
        this.closeSlideMenu();
        browserHistory.push(routes.home.childRoutes.main.path);
    }

    clickMyProfile(e){
        e.preventDefault();
        this.closeSlideMenu();
        browserHistory.push(routes.home.childRoutes.profile.path);
    }

    clickAccountSettings(e) {
        e.preventDefault();
        this.closeSlideMenu();
        browserHistory.push(routes.home.childRoutes.accountSettings.path);
    }

    closeSlideMenu() {
        this.setState({
            mobileSwitchOpen:false
        });
    }

    render() {
        return (
            <div className="navbar navbar-inverse navbar-fixed-top">
                <div className="navbar-boxed">
                    <div className="navbar-header">
                        <a className="navbar-brand" href="#"onClick={(e) => {
                                        e.preventDefault();
                                        browserHistory.push(routes.home.childRoutes.main.path);
                                    }}>
                            <div>GirishWeb</div>
                        </a>

                        <ul className="nav navbar-nav visible-xs-block">
                            <li>
                                <a className="sidebar-mobile-main-toggle">
                                    <i className="icon-search4"></i>
                                </a>
                            </li>
                            <li>
                                <a className="sidebar-mobile-main-toggle" onClick={()=>this.toggleDropdown('mobile-switch')}>
                                    <i className="icon-paragraph-justify3"></i>
                                </a>
                            </li>

                        </ul>

                        <VerticalSlideMenu openMenu={this.state.mobileSwitchOpen}>
                            <a href="#" onClick={(e) => this.clickHome(e)}><i className="icon-home2"></i><span>Home</span></a>
                            <a href="#" onClick={(e) => this.clickMyProfile(e)}><i className="icon-user-plus"></i><span>My Profile</span></a>
                            <a href="#" onClick={(e) => this.clickAccountSettings(e)}><i className="icon-cog5"></i><span>Account settings</span></a>
                            <a href="#" onClick={(e) => this.logoutClick(e)}><i className="icon-switch2"></i><span>Logout</span></a>
                        </VerticalSlideMenu>

                    </div>

                    <div className="navbar-collapse collapse" id="navbar-mobile" ref="component">
                        <ul className="nav navbar-nav navbar-right">
                            <li style={{display:'none'}} className={this.state.courseSwitchOpen ? "dropdown language-switch open":"dropdown language-switch"}
                                onClick={() => this.toggleDropdown('course-switch')}>
                                <a className="dropdown-toggle">

                                    <span>Strategic Marketing</span>
                                    <span className="caret"></span>
                                </a>

                                <ul className="dropdown-menu">
                                    <li>
                                        <a>Cognitive Science</a>
                                    </li>
                                    <li className="divider"></li>
                                    <li>
                                        <a>Computer Programming</a>
                                    </li>
                                </ul>
                            </li>


                            <li style={{display:'none'}} className={this.state.messagesSwitchOpen ? "dropdown open":"dropdown" }
                                onClick={() => this.toggleDropdown('messages-switch')}>
                                <a href="#" className="dropdown-toggle" aria-expanded={this.state.messagesSwitchOpen ? "true":"false" }>
                                    <i className="icon-bubbles4"></i>
                                    <span className="visible-xs-inline-block position-right">Messages</span>
                                    <span className="badge bg-warning-400">2</span>
                                </a>

                                <div className="dropdown-menu dropdown-content width-350">
                                    <div className="dropdown-content-heading">
                                        Messages
                                        <ul className="icons-list">
                                            <li><a href="#"><i className="icon-compose"></i></a></li>
                                        </ul>
                                    </div>

                                    <ul className="media-list dropdown-content-body">
                                        <li className="media">
                                            <div className="media-left">
                                                <img src="../assets/images/placeholder.jpg" className="img-circle img-sm" alt=""/>
                                                    <span className="badge bg-danger-400 media-badge">5</span>
                                                </div>

                                                <div className="media-body">
                                                    <a href="#" className="media-heading">
                                                        <span className="text-semibold">James Alexander</span>
                                                        <span className="media-annotation pull-right">04:58</span>
                                                    </a>

                                                    <span className="text-muted">who knows, maybe that would be the best thing for me...</span>
                                                </div>
                                            </li>

                                            <li className="media">
                                                <div className="media-left">
                                                    <img src="../assets/images/placeholder.jpg" className="img-circle img-sm" alt=""/>
                                                        <span className="badge bg-danger-400 media-badge">4</span>
                                                    </div>

                                                    <div className="media-body">
                                                        <a href="#" className="media-heading">
                                                            <span className="text-semibold">Margo Baker</span>
                                                            <span className="media-annotation pull-right">12:16</span>
                                                        </a>

                                                        <span className="text-muted">That was something he was unable to do because...</span>
                                                    </div>
                                                </li>

                                                <li className="media">
                                                    <div className="media-left">
                                                        <img src="../assets/images/placeholder.jpg" className="img-circle img-sm" alt=""/>
                                                    </div>

                                                        <div className="media-body">
                                                            <a href="#" className="media-heading">
                                                                <span className="text-semibold">Jeremy Victorino</span>
                                                                <span className="media-annotation pull-right">22:48</span>
                                                            </a>

                                                            <span className="text-muted">But that would be extremely strained and suspicious...</span>
                                                        </div>
                                                    </li>

                                                    <li className="media">
                                                        <div className="media-left">
                                                            <img src="../assets/images/placeholder.jpg" className="img-circle img-sm" alt=""/>
                                                        </div>

                                                            <div className="media-body">
                                                                <a href="#" className="media-heading">
                                                                    <span className="text-semibold">Beatrix Diaz</span>
                                                                    <span className="media-annotation pull-right">Tue</span>
                                                                </a>

                                                                <span className="text-muted">What a strenuous career it is that I've chosen...</span>
                                                            </div>
                                                        </li>

                                                        <li className="media">
                                                            <div className="media-left">
                                                                <img src="../assets/images/placeholder.jpg" className="img-circle img-sm" alt=""/>
                                                            </div>

                                                                <div className="media-body">
                                                                    <a href="#" className="media-heading">
                                                                        <span className="text-semibold">Richard Vango</span>
                                                                        <span
                                                                            className="media-annotation pull-right">Mon</span>
                                                                    </a>

                                                                    <span className="text-muted">Other travelling salesmen live a life of luxury...</span>
                                                                </div>
                                                            </li>
                                                        </ul>

                                                        <div className="dropdown-content-footer">
                                                            <a href="#" data-popup="tooltip" title=""
                                                               data-original-title="All messages"><i
                                                                className="icon-menu display-block"></i></a>
                                                        </div>
                                                    </div>
                                                </li>



                            <li className={this.state.userSwitchOpen ? "dropdown dropdown-user open":"dropdown dropdown-user"}
                                onClick={() => this.toggleDropdown('user-switch')}>
                                <a className="dropdown-toggle" data-toggle="dropdown">
                                    <img src="../assets/images/ron-carucci_avatar.jpg" alt=""/>
                                    <span>{this.props.currentUser.firstName}</span>
                                    <i className="caret"></i>
                                </a>

                                <ul className="dropdown-menu dropdown-menu-right">
                                    <li>
                                        <a href="#" onClick={(e) => this.clickHome(e)}>
                                            <i className="icon-home2"></i>Home
                                        </a>
                                    </li>
                                    <li>
                                        <a href="#" onClick={(e) => this.clickMyProfile(e)}>
                                            <i className="icon-user-plus"></i> My profile
                                        </a>
                                    </li>
                                    <li>
                                        <a href="#" onClick={(e) => this.clickAccountSettings(e)}>
                                            <i className="icon-cog5"></i> Account settings
                                        </a>
                                    </li>
                                    <li className="divider"></li>
                                    <li>
                                        <a href="#" onClick={(e) => this.logoutClick(e)}>
                                            <i className="icon-switch2"></i> Logout
                                        </a>
                                    </li>
                                </ul>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    //Whatever is returned from this function will end up
    // as props inside of this container component
    return {
        currentUser:state.currentUser,
        loggedIn:state.loggedIn
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({logoutUser,setHomeScreenLoading}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(HeaderContainer);