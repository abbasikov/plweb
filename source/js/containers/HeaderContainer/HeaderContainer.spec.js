import React from 'react';
import {shallow, render, mount } from 'enzyme';
import toJson  from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { INITIAL_STATE } from '../../constants/AppConstants';
import HeaderContainer from './HeaderContainer';

const mockStore = configureMockStore([ thunk ]);

describe('HeaderContainer', function() {
    var store;
    var container;

    beforeEach(function() {
        store = mockStore(INITIAL_STATE);
    });

    describe('#render', () => {
        it('initial state', function() {
            container = render(<Provider store={store}><HeaderContainer /></Provider>);
            expect(toJson(container)).toMatchSnapshot();
        });
    });

});