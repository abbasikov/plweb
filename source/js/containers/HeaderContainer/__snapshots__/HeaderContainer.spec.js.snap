// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer #render initial state 1`] = `
<div
  class="navbar navbar-inverse navbar-fixed-top"
>
  <div
    class="navbar-boxed"
  >
    <div
      class="navbar-header"
    >
      <a
        class="navbar-brand"
        href="#"
      >
        <div>
          VenueParking
        </div>
      </a>
      <ul
        class="nav navbar-nav visible-xs-block"
      >
        <li>
          <a
            class="sidebar-mobile-main-toggle"
          >
            <i
              class="icon-search4"
            />
          </a>
        </li>
        <li>
          <a
            class="sidebar-mobile-main-toggle"
          >
            <i
              class="icon-paragraph-justify3"
            />
          </a>
        </li>
      </ul>
      <nav
        class="cbp-spmenu cbp-spmenu-vertical cbp-spmenu-right"
      >
        <a
          href="#"
        >
          <i
            class="icon-home2"
          />
          <span>
            Home
          </span>
        </a>
        <a
          href="#"
        >
          <i
            class="icon-user-plus"
          />
          <span>
            My Profile
          </span>
        </a>
        <a
          href="#"
        >
          <i
            class="icon-cog5"
          />
          <span>
            Account settings
          </span>
        </a>
        <a
          href="#"
        >
          <i
            class="icon-switch2"
          />
          <span>
            Logout
          </span>
        </a>
      </nav>
    </div>
    <div
      class="navbar-collapse collapse"
      id="navbar-mobile"
    >
      <ul
        class="nav navbar-nav navbar-right"
      >
        <li
          class="dropdown language-switch"
        >
          <a
            class="dropdown-toggle"
          >
            <span>
              Strategic Marketing
            </span>
            <span
              class="caret"
            />
          </a>
          <ul
            class="dropdown-menu"
          >
            <li>
              <a>
                Cognitive Science
              </a>
            </li>
            <li
              class="divider"
            />
            <li>
              <a>
                Computer Programming
              </a>
            </li>
          </ul>
        </li>
        <li
          class="dropdown"
        >
          <a
            aria-expanded="false"
            class="dropdown-toggle"
            href="#"
          >
            <i
              class="icon-bubbles4"
            />
            <span
              class="visible-xs-inline-block position-right"
            >
              Messages
            </span>
            <span
              class="badge bg-warning-400"
            >
              2
            </span>
          </a>
          <div
            class="dropdown-menu dropdown-content width-350"
          >
            <div
              class="dropdown-content-heading"
            >
              Messages
              <ul
                class="icons-list"
              >
                <li>
                  <a
                    href="#"
                  >
                    <i
                      class="icon-compose"
                    />
                  </a>
                </li>
              </ul>
            </div>
            <ul
              class="media-list dropdown-content-body"
            >
              <li
                class="media"
              >
                <div
                  class="media-left"
                >
                  <img
                    alt=""
                    class="img-circle img-sm"
                    src="../assets/images/placeholder.jpg"
                  />
                  <span
                    class="badge bg-danger-400 media-badge"
                  >
                    5
                  </span>
                </div>
                <div
                  class="media-body"
                >
                  <a
                    class="media-heading"
                    href="#"
                  >
                    <span
                      class="text-semibold"
                    >
                      James Alexander
                    </span>
                    <span
                      class="media-annotation pull-right"
                    >
                      04:58
                    </span>
                  </a>
                  <span
                    class="text-muted"
                  >
                    who knows, maybe that would be the best thing for me...
                  </span>
                </div>
              </li>
              <li
                class="media"
              >
                <div
                  class="media-left"
                >
                  <img
                    alt=""
                    class="img-circle img-sm"
                    src="../assets/images/placeholder.jpg"
                  />
                  <span
                    class="badge bg-danger-400 media-badge"
                  >
                    4
                  </span>
                </div>
                <div
                  class="media-body"
                >
                  <a
                    class="media-heading"
                    href="#"
                  >
                    <span
                      class="text-semibold"
                    >
                      Margo Baker
                    </span>
                    <span
                      class="media-annotation pull-right"
                    >
                      12:16
                    </span>
                  </a>
                  <span
                    class="text-muted"
                  >
                    That was something he was unable to do because...
                  </span>
                </div>
              </li>
              <li
                class="media"
              >
                <div
                  class="media-left"
                >
                  <img
                    alt=""
                    class="img-circle img-sm"
                    src="../assets/images/placeholder.jpg"
                  />
                </div>
                <div
                  class="media-body"
                >
                  <a
                    class="media-heading"
                    href="#"
                  >
                    <span
                      class="text-semibold"
                    >
                      Jeremy Victorino
                    </span>
                    <span
                      class="media-annotation pull-right"
                    >
                      22:48
                    </span>
                  </a>
                  <span
                    class="text-muted"
                  >
                    But that would be extremely strained and suspicious...
                  </span>
                </div>
              </li>
              <li
                class="media"
              >
                <div
                  class="media-left"
                >
                  <img
                    alt=""
                    class="img-circle img-sm"
                    src="../assets/images/placeholder.jpg"
                  />
                </div>
                <div
                  class="media-body"
                >
                  <a
                    class="media-heading"
                    href="#"
                  >
                    <span
                      class="text-semibold"
                    >
                      Beatrix Diaz
                    </span>
                    <span
                      class="media-annotation pull-right"
                    >
                      Tue
                    </span>
                  </a>
                  <span
                    class="text-muted"
                  >
                    What a strenuous career it is that I've chosen...
                  </span>
                </div>
              </li>
              <li
                class="media"
              >
                <div
                  class="media-left"
                >
                  <img
                    alt=""
                    class="img-circle img-sm"
                    src="../assets/images/placeholder.jpg"
                  />
                </div>
                <div
                  class="media-body"
                >
                  <a
                    class="media-heading"
                    href="#"
                  >
                    <span
                      class="text-semibold"
                    >
                      Richard Vango
                    </span>
                    <span
                      class="media-annotation pull-right"
                    >
                      Mon
                    </span>
                  </a>
                  <span
                    class="text-muted"
                  >
                    Other travelling salesmen live a life of luxury...
                  </span>
                </div>
              </li>
            </ul>
            <div
              class="dropdown-content-footer"
            >
              <a
                data-original-title="All messages"
                data-popup="tooltip"
                href="#"
                title=""
              >
                <i
                  class="icon-menu display-block"
                />
              </a>
            </div>
          </div>
        </li>
        <li
          class="dropdown dropdown-user"
        >
          <a
            class="dropdown-toggle"
            data-toggle="dropdown"
          >
            <img
              alt=""
              src="../assets/images/ron-carucci_avatar.jpg"
            />
            <span />
            <i
              class="caret"
            />
          </a>
          <ul
            class="dropdown-menu dropdown-menu-right"
          >
            <li>
              <a
                href="#"
              >
                <i
                  class="icon-home2"
                />
                Home
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <i
                  class="icon-user-plus"
                />
                 My profile
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                <i
                  class="icon-cog5"
                />
                 Account settings
              </a>
            </li>
            <li
              class="divider"
            />
            <li>
              <a
                href="#"
              >
                <i
                  class="icon-switch2"
                />
                 Logout
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
